reference-controllers/rabbitmq-controller/src/main.rs

```diff
@@ -262,12 +250,18 @@ async fn restart_sts_if_needed(rabbitmq: &RabbitmqCluster, client: &Client) -> R
         .unwrap()
         .annotations
     {
-        Some(annotations) => annotations.get("restartedAt").cloned(),
+        Some(annotations) => annotations.get("restartedVersion").cloned(),
         None => None,
     };
 
+    let server_config_updatedat = server_config_o
+        .as_ref()
+        .unwrap()
+        .metadata
+        .resource_version
+        .clone();
     if sts_restartedat.is_some()
-        && sts_restartedat.clone().unwrap() > server_config_updatedat.clone().unwrap()
+        && sts_restartedat.clone().unwrap() >= server_config_updatedat.clone().unwrap()
```

Comment:
I don't think you have to check whether the version has changed -- if you update the statefulset with the exactly same annotation, there won't be any change and pods should not be restarted. But pls do some tests to validate what I said.